nix/master.nix

```
@@ -492,6 +492,75 @@ in
         '';
         default = 50;
       };
+
+      branches = lib.mkOption {
+        type = lib.types.attrsOf lib.types.submodule {
+          option = {
+            matchGlob = lib.mkOption {
+              type = lib.types.str;
+              description = ''
+                A glob specifying which branches to apply this rule to.
+              '';
+            };
+
+            registerGCRoots = lib.mkOption {
+              type = lib.types.bool;
+              description = ''
+                Whether to register gcroots for branches matching this glob.
+              '';
+              default = false;
+            };
+
+            copyOutputs = lib.mkOption {
+              type = lib.types.bool;
+              description = ''
+                Whether to copy outputs for branches matching this glob.
+              '';
+              default = false;
+            };
+          };
+        };
+        default = {};
+        description = ''
+          An attrset of branch rules, each rule specifies which branches it should apply to using the
+          `matchGlob` option and then the corresponding settings are applied to the matched branches.
+          If multiple rules match a given branch, the rules are `or`-ed together, by `or`-ing each
+          individual boolean option of all matching rules. Take the following as example:
+          ```
+             {
+               rule1 = {
+                 matchGlob = "f*";
+                 registerGCRroots = false;
+                 copyOutputs = false;
+               }
+               rule2 = {
+                 matchGlob = "foo";
+                 registerGCRroots = true;
+                 copyOutputs = false;
+               }
+             }
+          ```
+          This example will result in `build_prs` and `register_gcroots` both being considered `true`,
+          but `copy_outputs` being `false` for the branch `foo`.
+
+          The default branches of all repos are considered to be matching of a rule setting all the options
+          to `true`.
```

```suggestion
An attribute set of branch rules specifies how to apply settings to specific branches using the `matchGlob` option. Each rule defines which branches it applies to, and the corresponding settings are applied to all branches that match the pattern. If multiple rules apply to a given branch, the rules are combined by "or"-ing each individual boolean option across all matching rules. For example:

```
{
  rule1 = {
    matchGlob = "f*";
    registerGCRroots = false;
    copyOutputs = false;
  }
  rule2 = {
    matchGlob = "foo";
    registerGCRroots = true;
    copyOutputs = false;
  }
}
```

In this example, for the branch `foo`, both `build_prs` and `register_gcroots` are considered `true`, while `copy_outputs` remains `false`.

By default, the main branches of all repositories match a rule that sets all options to `true`.
```

---

https://gist.github.com/Mic92/9bdbd751ac1eb8200c8d5570aac0f936